src/file/format/mod.rs

```diff
@@ -81,6 +88,12 @@ lazy_static! {
         #[cfg(feature = "json5")]
         formats.insert(FileFormat::Json5, vec!["json5"]);
 
+        #[cfg(all(feature = "jsonc", feature = "json"))]
+        formats.insert(FileFormat::Jsonc, vec!["jsonc"]);
+
+        #[cfg(all(feature = "jsonc", not(feature = "json")))]
+        formats.insert(FileFormat::Jsonc, vec!["jsonc", "json"]);
```

Comment:
Do I understand correctly that the `jsonc` parser also parses `json` even if that feature is disabled?

I think that's counter-intuitive and we should only parse `jsonc` with the `jsonc` parser.